Std/Lean/Meta/DiscrTree.lean

```diff
@@ -19,11 +19,12 @@ Compare two `Key`s. The ordering is total but otherwise arbitrary. (It uses
 `Name.quickCmp` internally.)
 -/
 protected def cmp : Key s → Key s → Ordering
-  | .lit v₁,      .lit v₂      => compare v₁ v₂
-  | .fvar n₁ a₁,  .fvar n₂ a₂  => compareLex' Name.quickCmp n₁.name n₂.name compare a₁ a₂
-  | .const n₁ a₁, .const n₂ a₂ => compareLex' Name.quickCmp n₁ n₂ compare a₁ a₂
-  | .proj s₁ i₁,  .proj s₂ i₂  => compareLex' Name.quickCmp s₁ s₂ compare i₁ i₂
-  | k₁,           k₂           => compare k₁.ctorIdx k₂.ctorIdx
+  | .lit v₁,        .lit v₂         => compare v₁ v₂
+  | .fvar n₁ a₁,    .fvar n₂ a₂     => compareLex' Name.quickCmp n₁.name n₂.name compare a₁ a₂
+  | .const n₁ a₁,   .const n₂ a₂    => compareLex' Name.quickCmp n₁ n₂ compare a₁ a₂
+  | .proj s₁ i₁ _,  .proj s₂ i₂ _   => compareLex' Name.quickCmp s₁ s₂ compare i₁ i₂
+    -- TODO take a₁ and a₂ into account?
```

Comment:
This is for https://github.com/leanprover/lean4/commit/dd682bf1d5603a82fb63b861ace7958bc4032a63 - not sure what the right fix is

Reply:
You should take them into account, since the lt relation defined in that commit also does.

Reply:
Would you mind writing the code? Meta code isn't my strong suit